bundles/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/handler/EcobeeSensorThingHandler.java

```diff
@@ -0,0 +1,193 @@
+import org.eclipse.smarthome.core.library.unit.SmartHomeUnits;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.ThingStatusInfo;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.thing.binding.builder.ChannelBuilder;
+import org.eclipse.smarthome.core.thing.binding.builder.ThingBuilder;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.core.types.UnDefType;
+import org.openhab.binding.ecobee.internal.config.EcobeeSensorConfiguration;
+import org.openhab.binding.ecobee.internal.dto.thermostat.RemoteSensorCapabilityDTO;
+import org.openhab.binding.ecobee.internal.dto.thermostat.RemoteSensorDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link EcobeeSensorThingHandler} is responsible for updating the channels associated
+ * with an Ecobee remote sensor.
+ *
+ * @author Mark Hilbush - Initial contribution
+ */
+@NonNullByDefault
+public class EcobeeSensorThingHandler extends BaseThingHandler {
+
+    public static final String CAPABILITY_ADC = "adc";
+    public static final String CAPABILITY_CO2 = "co2";
+    public static final String CAPABILITY_DRY_CONTACT = "dryContact";
+    public static final String CAPABILITY_HUMIDITY = "humidity";
+    public static final String CAPABILITY_OCCUPANCY = "occupancy";
+    public static final String CAPABILITY_TEMPERATURE = "temperature";
+    public static final String CAPABILITY_UNKNOWN = "unknown";
+
+    private final Logger logger = LoggerFactory.getLogger(EcobeeSensorThingHandler.class);
+
+    private @NonNullByDefault({}) String sensorId;
+
+    private Map<String, State> stateCache = new ConcurrentHashMap<>();
+
+    public EcobeeSensorThingHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        sensorId = getConfigAs(EcobeeSensorConfiguration.class).sensorId;
+        logger.debug("SensorThing: Initializing sensor '{}'", sensorId);
+        clearSavedState();
+        updateStatus(EcobeeUtils.isBridgeOnline(getBridge()) ? ThingStatus.ONLINE : ThingStatus.OFFLINE);
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("SensorThing: Disposing sensor '{}'", sensorId);
+    }
+
+    @Override
+    public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
+        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @SuppressWarnings("null")
```

Comment:
What are you suppressing here?

Reply:
Eclipse complains about a redundant null check on the call `stateCache.get(channelUID.getId())`.